### ISSUE TITLE
How to specify taskRoleArn in the .yml file?

### ISSUE THREAD

**Issue by rclmenezes** [aws/amazon-ecs-cli#188]
The ecs TaskDefinition API allows you to specify the exact IAM role for your task:
http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_TaskDefinition.html

However, the docs don't specify if there's a way to do this in ecs-cli.

**Comment by kux** [comment]
+1 for this feature request

**Comment by kux** [comment]
In the meanwhile, a workaround is the fact that the containers inherit the policy that the role of the ec2 host has.

**Comment by ogrodnek** [comment]
@kux ec2 instance roles are better than nothing, but it's not really a workaround.  with ec2 roles all tasks running on that instance will have the host's role vs. task specific policies.

**Comment by BlizzardBlue** [comment]
+1

**Comment by mihbor** [comment]
+1

**Comment by vbudilov** [comment]
+1
@yinshiua @uttarasridhar 

Any idea if this is on the roadmap? It's an important security feature. 

There's also no way to specify the role on the EC2 instances themselves with *ecs-cli up* (right now I have to assign different roles or add policies after the instances have been provisioned), so it's a double whammy.

**Comment by yinshiua** [comment]
Hi, these are the features that we definitely want to support. We're still working through our priorities, and will keep you posted. Feel free to submit a PR.

**Comment by vbudilov** [comment]
Hi Jessica, a PR was submitted for this - I linked to our in my previous
comment.

**Comment by yinshiua** [comment]
I've commented on the PR :)

Feel free to submit a PR for instance role as well. Thanks!

**Comment by yinshiua** [comment]
You can now specify task role arn at runtime using `--task-role-arn`, but we will need a different solution as we add more ECS specific Task definition fields that are not supported by Docker compose in either the docker-compose yml or a separate ECS yaml.

**Comment by adenot** [comment]
> You can now specify task role arn at runtime using --task-role-arn, but we will need a different solution as we add more ECS specific Task definition fields that are not supported by Docker compose in either the docker-compose yml or a separate ECS yaml.

We could use labels in the compose yaml file to achieve this:

```
  labels:
    aws.ecs.task.role_arn: 'some_arn_here'
```

**Comment by jlyon** [comment]
@yinshiua `--task-role-arn` is not working for me in `:latest` (`VERSION:  0.6.0 (1053c82)`):
```
Incorrect Usage: flag provided but not defined: -task-role-arn
```
Do I need to build the binary locally? Based on the commit log, it looks like it should be in 0.6.0. Thanks!

**Comment by yinshiua** [comment]
Can you provide the command that you used?

**Comment by jlyon** [comment]
```
export SERVICE_ROLE=ecsServiceRole
export LOAD_BALANCER_ARN=xxx
export TASK_ROLE_ARN=xxx
ecs-cli compose service up \
    --role $SERVICE_ROLE \
    --target-group-arn $LOAD_BALANCER_ARN \
    --container-name apache \
    --container-port 80 \
    --task-role-arn $TASK_ROLE_ARN
```
With the ARNs set to the full ARN.

**Comment by yinshiua** [comment]
You need to apply the `--task-role-arn` after compose. See below:
```
ecs-cli compose --task-role-arn $TASK_ROLE_ARN service up
    --role $SERVICE_ROLE \
    --target-group-arn $LOAD_BALANCER_ARN \
    --container-name apache \
    --container-port 80 \
```

**Comment by jlyon** [comment]
Thanks, @yinshiua that did the trick!  I will need to set the `--task-role-arn` flag on over `ecs-cli compose` call, right (I can't just set it once on the service initialization like I am doing with `--target-group-arn` and `--role` because it lives at the Task Definition level)?

**Comment by yinshiua** [comment]
correct, Task Role ARN is used in the Task Definition, so you will need to apply it every time if you wish to use it.

**Comment by SoManyHs** [comment]
This was addressed in #328 and released in ECS v.1.0.0. Network mode can now be specified in the ecs-params.yml file as `task_role_arn`. Please let us know if you have any further issues!
